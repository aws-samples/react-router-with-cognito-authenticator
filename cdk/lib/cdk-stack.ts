import * as cdk from "@aws-cdk/core";
import * as cognito from "@aws-cdk/aws-cognito";

export class CdkStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const userPool = new cognito.UserPool(this, "UserPool", {
      selfSignUpEnabled: true,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });

    const userPoolClient = userPool.addClient("WebClient", {
      authFlows: {
        userPassword: true,
        userSrp: true,
      },
    });

    new cognito.CfnUserPoolGroup(this, "UserPoolGroupAdmin", {
      userPoolId: userPool.userPoolId,
      groupName: "admin",
    });

    new cdk.CfnOutput(this, "OutputUserPool", {
      value: userPool.userPoolId,
    });

    new cdk.CfnOutput(this, "OutputUserPoolClientId", {
      value: userPoolClient.userPoolClientId,
    });
  }
}
