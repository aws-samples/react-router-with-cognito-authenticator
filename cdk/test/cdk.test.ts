import * as cdk from "@aws-cdk/core";
import { Template } from "@aws-cdk/assertions";
import { CdkStack } from "../lib/cdk-stack";

test("snapshot test", () => {
  const app = new cdk.App();
  const stack = new CdkStack(app, "MyTestStack");
  // スタックからテンプレート(JSON)を生成
  const template = Template.fromStack(stack).toJSON();

  // 生成したテンプレートとスナップショットが同じか検証
  expect(template).toMatchSnapshot();
});
