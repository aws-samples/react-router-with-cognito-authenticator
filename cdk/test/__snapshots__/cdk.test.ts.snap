// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Outputs": Object {
    "OutputUserPool": Object {
      "Value": Object {
        "Ref": "UserPool6BA7E5F2",
      },
    },
    "OutputUserPoolClientId": Object {
      "Value": Object {
        "Ref": "UserPoolWebClient4C9370B0",
      },
    },
  },
  "Resources": Object {
    "UserPool6BA7E5F2": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "AccountRecoverySetting": Object {
          "RecoveryMechanisms": Array [
            Object {
              "Name": "verified_phone_number",
              "Priority": 1,
            },
            Object {
              "Name": "verified_email",
              "Priority": 2,
            },
          ],
        },
        "AdminCreateUserConfig": Object {
          "AllowAdminCreateUserOnly": false,
        },
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "VerificationMessageTemplate": Object {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Delete",
    },
    "UserPoolGroupAdmin": Object {
      "Properties": Object {
        "GroupName": "admin",
        "UserPoolId": Object {
          "Ref": "UserPool6BA7E5F2",
        },
      },
      "Type": "AWS::Cognito::UserPoolGroup",
    },
    "UserPoolWebClient4C9370B0": Object {
      "Properties": Object {
        "AllowedOAuthFlows": Array [
          "implicit",
          "code",
        ],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": Array [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin",
        ],
        "CallbackURLs": Array [
          "https://example.com",
        ],
        "ExplicitAuthFlows": Array [
          "ALLOW_USER_PASSWORD_AUTH",
          "ALLOW_USER_SRP_AUTH",
          "ALLOW_REFRESH_TOKEN_AUTH",
        ],
        "SupportedIdentityProviders": Array [
          "COGNITO",
        ],
        "UserPoolId": Object {
          "Ref": "UserPool6BA7E5F2",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
  },
}
`;
